modules/tinymce/src/plugins/link/main/ts/ui/DialogInfo.ts

```diff
@@ -17,15 +17,15 @@ const nonEmptyAttr = (dom: DOMUtils, elem: string | Element, name: string): Opti
   return val !== null && val.length > 0 ? Optional.some(val) : Optional.none();
 };
 
-const extractFromAnchor = (editor: Editor, anchor: HTMLAnchorElement): LinkDialogInfo['anchor'] => {
+const extractFromAnchor = (editor: Editor, anchor: Optional<HTMLAnchorElement>): LinkDialogInfo['anchor'] => {
   const dom = editor.dom;
   const onlyText = Utils.isOnlyTextSelected(editor);
   const text: Optional<string> = onlyText ? Optional.some(Utils.getAnchorText(editor.selection, anchor)) : Optional.none();
-  const url: Optional<string> = anchor ? Optional.some(dom.getAttrib(anchor, 'href')) : Optional.none();
-  const target: Optional<string> = anchor ? Optional.from(dom.getAttrib(anchor, 'target')) : Optional.none();
-  const rel = nonEmptyAttr(dom, anchor, 'rel');
-  const linkClass = nonEmptyAttr(dom, anchor, 'class');
-  const title = nonEmptyAttr(dom, anchor, 'title');
+  const url: Optional<string> = anchor.bind((anchor) => Optional.from(dom.getAttrib(anchor, 'href')));
+  const target: Optional<string> = anchor.bind((anchor) => Optional.from(dom.getAttrib(anchor, 'target')));
+  const rel = anchor.bind((anchor) => nonEmptyAttr(dom, anchor, 'rel'));
+  const linkClass = anchor.bind((anchor) => nonEmptyAttr(dom, anchor, 'class'));
+  const title = anchor.bind((anchor) => nonEmptyAttr(dom, anchor, 'title'));
```

Comment:
You've got variable name shadowing here now which we shouldn't do (we unfortunately had to disable the linter rule due to pre-existing issues). Can you please come up with a different name either in `bind` or for the original variable.

Reply:
Done!